### PR TITLE
Renaming json to jsonl

### PR DESCRIPTION
**Pull request by xLaszlo**

### CODE REVIEW

File: src/hypergol/dataset.py [hunk 1]
```diff
@@ -1,21 +1,16 @@
...
-import json
-import hashlib
 
-from datetime import datetime
 from pathlib import Path
 
+from hypergol.datachunk import DataChunk
 from hypergol.utils import Repr
-from hypergol.base_data import BaseData
+from hypergol.utils import _get_hash
```
Comment (PattenR) [inline src/hypergol/dataset.py:8]: Importing a private function from another file? Something is weird here

File: src/hypergol/dataset.py [hunk 2]
```diff
@@ -26,140 +21,11 @@ class DatasetAlreadyExistsException(Exception):
...
-        for line in self.file:
-            yield self.dataset.dataType.from_data(json.loads(line.rstrip()))
-
-
 class Dataset(Repr):
     """
     Dataset class to store BaseData objects that is readable/writable in a parallel manner.
 
-    Files will be stored in: ``location/project/branch/name/name_???.json.gz``
+    Files will be stored in: ``location/project/branch/name/name_???.jsonl.gz``
```
Comment (PattenR) [inline src/hypergol/dataset.py:28]: This is an odd change?

File: src/hypergol/dataset_chk_file.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        chkFileData = json.loads(open(self.chkFilename, 'rt').read())
+        mv = memoryview(bytearray(CHECKSUM_BUFFER_SIZE))
+        for fileName, chkFileChecksum in chkFileData.items():
+            if fileName.endswith('.def'):
+                data = open(self.dataset.defFile.defFilename, 'rt').read()
+                actualChecksum = _get_hash(data)
+            else:
+                hasher = hashlib.sha1(''.encode('utf-8'))
+                with gzip.open(f'{self.dataset.directory}/{fileName}', 'rb') as f:
+                    for n in iter(lambda: f.readinto(mv), 0):   # pylint: disable=cell-var-from-loop
```
Comment (PattenR) [inline src/hypergol/dataset_chk_file.py:54]: I don’t get why this needs to be updated on every step? If different regions of memory were done each time then sure but aren’t you writing over each step?

File: src/hypergol/dataset_factory.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        chunkCount : int=None
+            Number of chunks, if None, the factory's own value will be used
+        """
+        return Dataset(
+            dataType=dataType,
+            location=self.location,
+            project=self.project,
+            branch=self.branch,
+            name=name,
+            chunkCount=chunkCount or self.chunkCount,
```
Comment (PattenR) [inline src/hypergol/dataset_factory.py:49]: Really this should be 

Chunkcout if chunkcount is not none else self.chunkcount

Although chunkcount=0 is a bit weird so don’t mind

File: src/hypergol/dataset.py (hunk 1 above)
Comment (xLaszlo) [reply src/hypergol/dataset.py:8]: yeah, should have been refactored to be public

File: src/hypergol/dataset.py (hunk 2 above)
Comment (xLaszlo) [reply src/hypergol/dataset.py:28]: yes, the format we use is jsonl, each line is a separate json, so we are just aligning with the right terminology. JSONL is a thing.

File: src/hypergol/dataset_chk_file.py (hunk 3 above)
Comment (xLaszlo) [reply src/hypergol/dataset_chk_file.py:54]: stackoverflow power, I look again into it tomorrow but I must admit I have no idea what goes on here

**Pull request merged**